sstables/sstables.hh

```diff
@@ -417,6 +417,17 @@ public:
 
     bool requires_view_building() const noexcept { return _state == sstable_state::staging; }
 
+    future<> wait_until_view_is_built() {
+        // wait for sstable to move out of staging, which is done after the view is updated.
+        if (_state != sstable_state::staging) {
```

Comment:
might want to use the accessor of `require_view_building()` instead. for better readability, as its name indicates that the state is connected with "view building".

Reply:
also, i am wondering, since we move sstables from "staging" to "normal" once they are processed. shall we instead check if the state is `sstable_state::normal` instead ? just for better readability. as what we expect is just a "normal" sstable.

Reply:
I have changed the approach in the new version.